src/Concerns/CapturesState.php

```diff
@@ -293,6 +298,27 @@ public function prepareForJob(Job $job): void
         }
     }
 
+    /**
+     * @internal
+     */
+    public function captureArtisan(Artisan $artisan): void
+    {
+        /** @var Core<CommandState> $this */
+        $this->state->artisan = $artisan;
+    }
```

Comment:
This happens before we have configured the sample rate, so we will always run this.